api/jobs.go

```diff
@@ -1643,3 +1643,37 @@ type JobStatusesRequest struct {
 	// IncludeChildren will include child (batch) jobs in the response.
 	IncludeChildren bool
 }
+
+type TagVersionRequest struct {
+	Version     uint64
+	Description string
+	WriteRequest
+}
+
+func (j *Jobs) TagVersion(jobID string, version string, name string, description string, q *WriteOptions) (*WriteMeta, error) {
+	// If the version is not provided, get the "active" version of the job
+	var versionInt uint64
+	if version == "" {
+		job, _, err := j.Info(jobID, nil)
+		if err != nil {
+			return nil, err
+		}
+		versionInt = *job.Version
+	} else {
+		var parseErr error
+		versionInt, parseErr = strconv.ParseUint(version, 10, 64)
+		if parseErr != nil {
+			return nil, fmt.Errorf("invalid version format: %q. Version must be a positive integer", version)
+		}
+	}
```

Comment:
My earlier comment here got resolved because of the pointer change (which looks great), but I think this bit still applies:

> Either way I would definitely require this function take in a uint64 though. Consider that callers outside the CLI (ex. the Terraform provider) will likely already have a uint64 from the Job object so forcing them to use a string here just makes for unnecessary conversions. Use the type system to enforce correctness here rather than parsing.

Reply:
My worry about using a uint64 here is that it's non-nilable, and `0` is a valid job version to pass in;

I want users to be able to not pass in a job version, and for us to infer they want the latest version to be tagged; I haven't been able to figure out how to do that with a uint64 unless it's a pointer instead.

Reply:
Sure, but I wouldn't do that here in the Go API. Do that in the caller.